File: app/src/main/java/com/elyeproj/javadaggeruitest/ApplicationModule.java
package com.elyeproj.javadaggeruitest;

import javax.inject.Singleton;

import dagger.Module;
import dagger.Provides;

@Module
public class ApplicationModule {

    @Provides
    @Singleton
    InjectedData provideInjectedData() {
        return new InjectedData();
    }
}

File: app/src/main/java/com/elyeproj/javadaggeruitest/ApplicationModule.kt
package com.elyeproj.javadaggeruitest

import android.app.Application

import javax.inject.Singleton

import dagger.Module
import dagger.Provides

@Module
class ApplicationModule(val mApplication: Application) {

    @Provides
    internal fun provideApplication(): Application {
        return mApplication
    }

    @Provides
    @Singleton
    internal fun provideInjectedData(): InjectedData {
        return InjectedData()
    }
}
